File: platform_src/android/app/src/main/java/org/elastos/trinity/runtime/passwordmanager/PasswordGetInfoOptions.java
package org.elastos.trinity.runtime.passwordmanager;

import org.json.JSONObject;

public class PasswordGetInfoOptions {
    public boolean promptPasswordIfLocked = true;
    public boolean forceMasterPasswordPrompt = false;

    public PasswordGetInfoOptions() {
    }

    public static PasswordGetInfoOptions fromJsonObject(JSONObject jsonObject) throws Exception {
        PasswordGetInfoOptions options = new PasswordGetInfoOptions();

        if (jsonObject.has("promptPasswordIfLocked"))
            options.promptPasswordIfLocked = jsonObject.getBoolean("promptPasswordIfLocked");

        if (jsonObject.has("forceMasterPasswordPrompt"))
            options.forceMasterPasswordPrompt = jsonObject.getBoolean("forceMasterPasswordPrompt");

        return options;
    }
}

File: platform_src/electron/main/passwordmanager/PasswordGetInfoOptions.ts
export class PasswordGetInfoOptions {
    public promptPasswordIfLocked: boolean = true;
    public forceMasterPasswordPrompt: boolean = false;

    public PasswordGetInfoOptions() {
    }

    public static fromJsonObject(jsonObject: any): PasswordGetInfoOptions {
         let options: PasswordGetInfoOptions = new PasswordGetInfoOptions();

        if (jsonObject.promptPasswordIfLocked != null)
            options.promptPasswordIfLocked = jsonObject.promptPasswordIfLocked as boolean;

        if (jsonObject.forceMasterPasswordPrompt != null)
            options.forceMasterPasswordPrompt = jsonObject.forceMasterPasswordPrompt as boolean;

        return options;
    }
}